lib/loopback.js

```diff
@@ -76,6 +95,10 @@ function createApplication() {
   app.connector('memory', loopback.Memory);
   app.connector('remote', loopback.Remote);
 
+  // setup the app registry
+  var registry = app.registry = new Registry();
+  require('./builtin-models')(registry);
+
```

Comment:
Also the flag implementation has to be clever, because the `app.registry` object is initialised before `app.set` can be called.

However, I still think it is a good idea to let the application decide whether it wants local or global model registry.

A possible solution if we want to keep the flag configurable via `config.json`:
 - `var app = loopback()` initialises `app.registry = loopback.registry`
 - `app.set('flag', true)` is intercepted and calls `app.registry = new Registry()`

I don't like the fact that if you happen to register models between the line where you create the app and the line where you switch to per-app registry, these models will be "lost".

A better approach is to keep the local/global registry switch in the code.

```js
// Option A - add "options" object, though it's rather un-express-like :(
var app = loopback({ localRegistry: true });

// Option B - add another factory method
var app = loopback.isolated();

loopback.isolated = function() {
  var app = loopback();
  app.registry = new Registry();
  return app;
}
```